### ISSUE TITLE
Does this work with Ecto 2.0.0-beta.1?

### ISSUE THREAD

**Issue by dre1080** [Dania02525/apartmentex#2]

**Comment by Dania02525** [comment]
This is currently set up for pre-2.0 Ecto, since It creates its own process from which to run migrations.  In Ecto 2.0, this library will end up much simpler since many of the structural enhancements have been since merged into ecto 2.0.  

I'll leave this issue open and will create a branch for ecto 2.0 support- which will basically be a stripped down version of this.

**Comment by andreapavoni** [comment]
Hi there! Is the project still alive?

**Comment by Dania02525** [comment]
@apeacox Still alive- just waiting for Ecto 2.0 to sort out all changes, and I will branch this to work properly for it.  If there are any issues you are experiencing using it with pre 2.0, please let me know!

**Comment by andreapavoni** [comment]
Thank you for the reply :-)

to be honest I'll need this package in the near future, I'm already using Ecto v2.0.0-rc5, so I was just wondering if the project was still alive :-P

BTW, I'll file eventual issues (and try to fixe them accordingly) if I'll found any :-)

thank you again!

**Comment by iancanderson** [comment]
Hello! I'm looking into maybe using this on an Ecto 2 project. Any news on the upgrade front now that Ecto 2.0 is out? I'm also curious which changes made it into Ecto that will simplify this library. Thanks!

**Comment by Dania02525** [comment]
@iancanderson Yes, I've been waiting on on the Phoenix RC to include this, to ensure beginners experience minimal frustration when bringing this library into a new phoenix project.  I'm going to go ahead and test it out with a new phoenix/ecto 2.0 project, and pull out the unneeded stuff in the library.  

The changes made to ecto are actually having to to with ecto itself having a process to manage migrations, so this library no longer needs to start one (it just needs to use the ecto one).  I haven't kept up on the changes to the repo interface with ecto 2.0, so there may be some other minor changes to make to apartmentex to ensure it works.

**Comment by iancanderson** [comment]
@Dania02525 I starting looking into upgrading this to Ecto 2.0, and it looks like a lot of code had been copied from Ecto originally and slightly modified. Would you be able to point me towards some of the more important bits where the custom multitenancy behavior lives?

**Comment by Dania02525** [comment]
@iancanderson The main part of this library is wrapping of existing ecto functionality to make it easier to work with schemas in PG and databases in MySql- with the exception of `new_tenant/2` and `drop_tenant/2` which send sql to do things that previous ecto versions can't do natively.  

The main differentiating functionality is within lib/migration/manager, and around the migrations behavior.  An supervised process is provided which holds the state of migrations in process instead of simply the migration agent being a single process for all migrations as in earlier versions of ecto.  This allows multiple tenant migrations to run in parallel (say, when two tenants sign up at the same time).  I made some commits to Ecto around this migration manager which were merged in, so Apartmentex should now be able to use the Ecto provided agent instead of needing to create its own.  Here is the Ecto provided agent: https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/migration/supervisor.ex

**Comment by iancanderson** [comment]
@Dania02525 thanks for the background. I took another stab at it, this time, bypassing the apartmentex migration code in favor of Ecto 2.0's.

I think this commit proves that it's possible to delete a bunch of code from apartmentex: https://github.com/Dania02525/apartmentex/commit/0784f88d2dd4ce572c15e4d2c3b9b9293148b7bd

Is this what you had in mind for the upgrade? There's a lot more that has to be done, but I wanted to get your general thoughts at this point.

**Comment by Dania02525** [comment]
Yes, this is exactly the direction I was thinking that we'd need to head.  I actually think lots of the other code can be deleted too, since it had to do with getting the migration running.  We essentially just need the apartmentex.ex file to manage the ecto functions that exist in 2.0, since we have no need to reopen the adapter modules or start a migration manager with 2.0.  I'll scan through when I get some time this week to have a look at what if anything we still need in /lib.

**Comment by iancanderson** [comment]
@Dania02525 glad to hear! Check out the PR when you get a chance 😄 https://github.com/Dania02525/apartmentex/pull/10
